### ISSUE TITLE
Missing public for aws_ami and aws_ami_from_instance

### ISSUE THREAD

**Issue by cadavre** [hashicorp/terraform-provider-aws#11101]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently there's no way to mark created AMI image as available public. Proposal is to add `public` parameter that defaults to `false`. Setting to true would mark AMI public and accessible for all.

### New or Affected Resource(s)

* aws_ami
* aws_ami_from_instance
* aws_ami_copy (nice-to-have)

### Potential Terraform Configuration

```hcl
resource "aws_ami_from_instance" "my_ami" {
  name               = "my-ami"
  source_instance_id = aws_instance.ami_source.id
  public             = true
}
```

**Comment by atz** [comment]
I think this implies reworking `aws_ami_launch_permission` to accept more than just an account_id (specifically `public`).  Or is it the (undocumented) case that `aws_ami_launch_permission` already accepts `public` as an account_id?

**Comment by grahamc** [comment]
I've implemented this in #20677, and you can use it pretty easily if you're using Nix to manage your Terraform environment. See: https://github.com/hashicorp/terraform-provider-aws/pull/20677#issuecomment-905694793
